pipeline/utils/github_utils.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utility functions related to GitHub interaction. Uses the GitHub API with
+reference https://developer.github.com/v3/"""
+
+import requests
+import json
+import os
+
+_API_URL = 'https://api.github.com'
+# file modes for different git file types
+_TREE_EL_MODES = {'file': u'100644',
+                  'executable': u'100755',
+                  'tree': u'040000',
+                  'symlink': u'120000'}
+_REQUEST_FUNCS = {'GET': requests.get,
```

Comment:
Not sure that this map is really useful -- it requires using string values (e.g., line 104) below where it would be simpler just to pass `requests.XXX` as a parameter directly.